observatory-platform/observatory/platform/cli/generate_command.py

```diff
@@ -73,95 +64,135 @@ def generate_terraform_config(self, config_path: str):
+                                    telescope_class=telescope_class,
+                                    author_name=author_name)
+        test = render_template(test_path,
+                               telescope_module=telescope_module,
+                               telescope_class=telescope_class,
+                               author_name=author_name)
+        doc = render_template(doc_path,
+                              telescope_module=telescope_module,
+                              telescope_class=telescope_class,
+                              )
+        schema = render_template(schema_path)
+
+        # Get paths to files
+        observatory_dir = module_file_path('observatory.platform', nav_back_steps=-4)
+
+        dag_dst_dir = module_file_path('observatory.dags.dags')
+        dag_dst_file = os.path.join(dag_dst_dir, f"{telescope_module}.py")
+
+        telescope_dst_dir = module_file_path('observatory.dags.telescopes')
+        telescope_dst_file = os.path.join(telescope_dst_dir, f"{telescope_module}.py")
+
+        test_dst_dir = os.path.join(observatory_dir, 'tests', 'observatory', 'dags', 'telescopes')
         test_dst_file = os.path.join(test_dst_dir, f"test_{telescope_module}.py")
 
-        doc_dst_dir = "docs"
-        doc_dst_file = os.path.join(doc_dst_dir, "telescopes", f"{telescope_module}.md")
+        doc_dst_dir = os.path.join(observatory_dir, 'docs', 'telescopes')
+        doc_dst_file = os.path.join(doc_dst_dir, f"{telescope_module}.md")
 
-        doc_index_file = os.path.join(doc_dst_dir, "telescopes", "index.rst")
+        schema_dst_dir = module_file_path('observatory.dags.database.schema')
+        schema_dst_file = os.path.join(schema_dst_dir, f"{telescope_module}_{datetime.now().strftime('%Y-%m-%d')}.json")
 
-        print(f"Created a new dag file: {dag_dst_file}")
-        print(f"Created a new telescope file: {telescope_dst_file}")
-        print(f"Created a new telescope documentation file: {doc_dst_file}")
-        print(f"Created a new telescope test file: {test_dst_file}")
+        # Write out files
+        write_telescope_file(dag_dst_file, dag, "dag")
+        write_telescope_file(telescope_dst_file, telescope, "telescope")
+        write_telescope_file(test_dst_file, test, "test")
+        write_telescope_file(doc_dst_file, doc, "documentation")
+        write_telescope_file(schema_dst_file, schema, "schema")
 
-        # Write out dag template
-        with open(dag_dst_file, "w") as f:
-            f.write(dag)
+        # Update documentation index
+        doc_index_file = os.path.join(doc_dst_dir, "index.rst")
+        with open(doc_index_file, "a") as f:
+            f.write(f"    {telescope_module}\n")
+        print(f"- Updated the documentation index file: {doc_index_file}")
 
-        # Write out telescope template
-        with open(telescope_dst_file, "w") as f:
-            f.write(telescope)
+        # Update TelescopeTypes in identifiers.py when using organisation template
```

Comment:
can this also be handled by the write_telescope_file function?

Reply:
It's not as much writing a file, but updating an existing file and only for a specific telescope type.
So I would say the functionality is a bit different.